rfcs/0004-categorized-attributes.md

```diff
@@ -0,0 +1,217 @@
+- Feature Name: Categorized attributes
+- Affected audience: Game Developers
+- RFC PR: [#9114](https://github.com/endless-sky/endless-sky/pull/9114)
+- Relevant Issues/RFCs: [#9025](https://github.com/endless-sky/endless-sky/pull/9025)
+
+# Summary
+
+Provides a way of programmatically accessing attributes instead of requiring the unique, string-form name of the attribute.
```

Comment:
This text (and some later texts) are not very clear on what we want to achieve. (Let's look at the other review comments first, and handle this review-comment last; the referenced Issues give a reasonable explanation on what we want.)